internal/service/flexcluster/resource_schema.go

```diff
@@ -0,0 +1,187 @@
+)
+
+func ResourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"project_id": schema.StringAttribute{
+				Required:            true,
+				Description:         "Unique 24-hexadecimal character string that identifies the project.",
+				MarkdownDescription: "Unique 24-hexadecimal character string that identifies the project.",
+			},
+			"name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Human-readable label that identifies the instance.",
+				MarkdownDescription: "Human-readable label that identifies the instance.",
+			},
+			"provider_settings": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"backing_provider_name": schema.StringAttribute{
+						Required:            true,
+						Description:         "Cloud service provider on which MongoDB Cloud provisioned the flex cluster.",
+						MarkdownDescription: "Cloud service provider on which MongoDB Cloud provisioned the flex cluster.",
+					},
+					"disk_size_gb": schema.Float64Attribute{
+						Computed:            true,
+						Description:         "Storage capacity available to the flex cluster expressed in gigabytes.",
+						MarkdownDescription: "Storage capacity available to the flex cluster expressed in gigabytes.",
+					},
+					"provider_name": schema.StringAttribute{
+						Computed:            true,
+						Description:         "Human-readable label that identifies the cloud service provider.",
+						MarkdownDescription: "Human-readable label that identifies the cloud service provider.",
+					},
+					"region_name": schema.StringAttribute{
+						Required:            true,
+						Description:         "Human-readable label that identifies the geographic location of your MongoDB flex cluster. The region you choose can affect network latency for clients accessing your databases. For a complete list of region names, see [AWS](https://docs.atlas.mongodb.com/reference/amazon-aws/#std-label-amazon-aws), [GCP](https://docs.atlas.mongodb.com/reference/google-gcp/), and [Azure](https://docs.atlas.mongodb.com/reference/microsoft-azure/).",
+						MarkdownDescription: "Human-readable label that identifies the geographic location of your MongoDB flex cluster. The region you choose can affect network latency for clients accessing your databases. For a complete list of region names, see [AWS](https://docs.atlas.mongodb.com/reference/amazon-aws/#std-label-amazon-aws), [GCP](https://docs.atlas.mongodb.com/reference/google-gcp/), and [Azure](https://docs.atlas.mongodb.com/reference/microsoft-azure/).",
+					},
+				},
+				Required:            true,
+				Description:         "Group of cloud provider settings that configure the provisioned MongoDB flex cluster.",
+				MarkdownDescription: "Group of cloud provider settings that configure the provisioned MongoDB flex cluster.",
+			},
+			"tags": schema.ListNestedAttribute{
```

Comment:
I believe it will be better to define tags as a Map as we did in project resource (taking into account benefits we found there cc: @EspenAlbert). In terms of automation not a big fan of this though, as it will require more configuration/customization.

Reply:
+1 On the tags as Map. Believe it is a better user experience and we should be consistent about it.

Reply:
we might even extract Tags map to a common file as it can be used for all resources

Reply:
Changed it to map